### ISSUE TITLE
Missing file name when emiting warnings

### ISSUE THREAD

**Issue by matejsp** [astral-sh/ruff#5729]
```
(venv) ➜  rm -rf .ruff_cache && ruff .
warning: Invalid `# noqa` directive on line 6: expected a comma-separated list of codes (e.g., `# noqa: F401, F841`).
warning: Invalid `# noqa` directive on line 1: expected a comma-separated list of codes (e.g., `# noqa: F401, F841`).
warning: Invalid `# noqa` directive on line 7: expected a comma-separated list of codes (e.g., `# noqa: F401, F841`).
warning: Invalid `# noqa` directive on line 3: expected a comma-separated list of codes (e.g., `# noqa: F401, F841`).
warning: Invalid `# noqa` directive on line 25: expected a comma-separated list of codes (e.g., `# noqa: F401, F841`).
```

This warning is emitted only on first run (after cache is build it is no longer displayed)
It would be nice to threat warnings as errors

Ruff version:
(venv) ➜ ruff --version
ruff 0.0.27

Previous version didn't emit such warnings on lines:
``` # noqa: 901 ```
now it wants complete code:
``` # noqa: C901 ```

**Comment by zanieb** [comment]
@matejsp just to clarify, you're looking for three things here?

1. Include the file name in the warnings
2. Always emit warnings, even if the cache exists
3. The ability to error on warnings

Is that right?

**Comment by matejsp** [comment]
1. YES (bug title)
2. would be nice
3. would also be nice :D

**Comment by charliermarsh** [comment]
I think the "right" solution here might be to treat those warnings as diagnostics, rather than "warnings". As diagnostics, they would have all of those properties automatically.

**Comment by charliermarsh** [comment]
It's similar to #850.

**Comment by charliermarsh** [comment]
Maybe I'll rephrase that as: they should very likely be diagnostics, and not ad hoc warnings. But the downside of being diagnostics is that they won't be enabled by default for users.
